### ISSUE TITLE
IPython seems to be crashing on some basic boto3 operations - Python 3.6 on Mac

### ISSUE THREAD

**Issue by buchs** [ipython/ipython#12986]
I needed to write code to support a user who has Python 3.6 installed (I know it is EOL). I just returned to working on a Mac, after a couple years away, so I may have missed something obvious. I use homebrew for Python 3.9. However, to install 3.6 I used the Mac installer from Python.com. I installed IPython using the pip tool that was installed with Python 3.6. I am running MacOS Big Sur. The most recent failure I have encountered, like the others I've seen, blew me right out of IPython and back to the shell. I tried the same operations in Idle and it was successful. Here is the transcript and traceback of that latest failure:
```
$ ipython3.6
Python 3.6.8 (v3.6.8:3c6b436a57, Dec 24 2018, 02:04:31)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import boto3

In [2]: boto3.client?
Signature: boto3.client(*args, **kwargs)
Docstring:
Create a low-level service client by name using the default session.

See :py:meth:`boto3.session.Session.client`.
File:      /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/boto3/__init__.py
Type:      function

In [3]: from botocore.config import Config

In [4]: config = Config(region_name='us-west-2')

In [5]: ecs = boto3.client('ecs', config=config)

In [6]: ecs.list_clusters()
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/ptutils.py", line 113, in get_completions
    yield from self._get_completions(body, offset, cursor_position, self.ipy_completer)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/ptutils.py", line 126, in _get_completions
    for c in completions:
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/core/completer.py", line 438, in _deduplicate_completions
    completions = list(completions)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/core/completer.py", line 1818, in completions
    for c in self._completions(text, offset, _timeout=self.jedi_compute_type_timeout/1000):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/core/completer.py", line 1862, in _completions
    full_text=full_text, cursor_line=cursor_line, cursor_pos=cursor_column)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/core/completer.py", line 2030, in _complete
    cursor_pos, cursor_line, full_text)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/core/completer.py", line 1374, in _jedi_matches
    text[:offset], namespaces, column=cursor_column, line=cursor_line + 1)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/jedi/api/__init__.py", line 726, in __init__
    project=Project(Path.cwd()), **kwds)
TypeError: __init__() got an unexpected keyword argument 'column'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/kevinbuchs/bin/ipython3.6", line 11, in <module>
    sys.exit(start_ipython())
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/__init__.py", line 126, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/traitlets/config/application.py", line 664, in launch_instance
    app.start()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/ipapp.py", line 356, in start
    self.shell.mainloop()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/interactiveshell.py", line 563, in mainloop
    self.interact()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/interactiveshell.py", line 546, in interact
    code = self.prompt_for_code()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/interactiveshell.py", line 474, in prompt_for_code
    **self._extra_prompt_options())
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/shortcuts/prompt.py", line 1021, in prompt
    set_exception_handler=set_exception_handler, in_thread=in_thread
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/application/application.py", line 888, in run
    self.run_async(pre_run=pre_run, set_exception_handler=set_exception_handler)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 484, in run_until_complete
    return future.result()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/application/application.py", line 816, in run_async
    return await _run_async2()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/application/application.py", line 798, in _run_async2
    await self.cancel_and_wait_for_background_tasks()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/application/application.py", line 945, in cancel_and_wait_for_background_tasks
    await task
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/buffer.py", line 1909, in new_coroutine
    await coroutine(*a, **kw)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/buffer.py", line 1739, in async_completer
    document, complete_event
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/completion/base.py", line 272, in get_completions_async
    document, complete_event
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/prompt_toolkit/completion/base.py", line 198, in get_completions_async
    for item in self.get_completions(document, complete_event):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/IPython/terminal/ptutils.py", line 116, in get_completions
    print_tb(e)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/traceback.py", line 53, in print_tb
    print_list(extract_tb(tb, limit=limit), file=file)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/traceback.py", line 72, in extract_tb
    return StackSummary.extract(walk_tb(tb), limit=limit)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/traceback.py", line 345, in extract
    for f, lineno in frame_gen:
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/traceback.py", line 310, in walk_tb
    yield tb.tb_frame, tb.tb_lineno
AttributeError: 'TypeError' object has no attribute 'tb_frame'

If you suspect this is an IPython 7.16.1 bug, please report it at:
    https://github.com/ipython/ipython/issues
or send an email to the mailing list at ipython-dev@python.org

You can print a more detailed traceback right now with "%tb", or use "%debug"
to interactively debug it.

Extra-detailed tracebacks for bug-reporting purposes can be enabled via:
    %config Application.verbose_crash=True

$ 
```

**Comment by buchs** [comment]
After setting Application.verbose_crash=True, I repeated the above steps and got more details, attached here.
[Crash_report_ipython.txt](https://github.com/ipython/ipython/files/6519040/Crash_report_ipython.txt)

**Comment by MrMino** [comment]
Have you tried to reproduce it on a newer version of Python? 3.6 is quite old, and no longer supported by IPython. I don't have a Mac, so I cannot reproduce it myself.

**Comment by buchs** [comment]
I usually use the latest Python version, but I had create something for someone who was only running 3.6. As I explained, I installed 3.6 via the installer downloaded from python.org. I have python3.9 installed by Homebrew. With that version, the problem does not occur. Note it runs IPython 7.23.1. I don't want to mess up my 3.9 install by trying the downloadable installer from Python.org. I may be getting my machine replaced in a few days and if so, I'll try to remember to give it a try on the new or old system.
